oauth2/fosite_store_helpers.go

```diff
@@ -52,28 +53,80 @@ var defaultRequest = fosite.Request{
+		Client:         &client.Client{ClientID: "foobar"},
+		RequestedScope: fosite.Arguments{"fa", "ba"},
+		GrantedScope:   fosite.Arguments{"fa", "ba"},
+		Form:           url.Values{"foo": []string{"bar", "baz"}},
+		Session:        &Session{DefaultSession: &openid.DefaultSession{Subject: "bar"}},
+	},
+	{
+		ID:             "flush-2",
+		RequestedAt:    time.Now().Round(time.Second).Add(-(lifespan + time.Minute)),
+		Client:         &client.Client{ClientID: "foobar"},
+		RequestedScope: fosite.Arguments{"fa", "ba"},
+		GrantedScope:   fosite.Arguments{"fa", "ba"},
+		Form:           url.Values{"foo": []string{"bar", "baz"}},
+		Session:        &Session{DefaultSession: &openid.DefaultSession{Subject: "bar"}},
+	},
+	{
+		ID:             "flush-3",
+		RequestedAt:    time.Now().Round(time.Second).Add(-(lifespan + time.Hour)),
+		Client:         &client.Client{ClientID: "foobar"},
+		RequestedScope: fosite.Arguments{"fa", "ba"},
+		GrantedScope:   fosite.Arguments{"fa", "ba"},
+		Form:           url.Values{"foo": []string{"bar", "baz"}},
+		Session:        &Session{DefaultSession: &openid.DefaultSession{Subject: "bar"}},
+	},
+}
+
+func mockRequestForeignKey(t *testing.T, id string, x ManagerTestSetup, createClient bool) {
 	cl := &client.Client{ClientID: "foobar"}
 	cr := &consent.ConsentRequest{
 		Client: cl, OpenIDConnectContext: new(consent.OpenIDConnectContext), LoginChallenge: id,
 		Challenge: id, Verifier: id, AuthenticatedAt: time.Now(), RequestedAt: time.Now(),
 	}
 
 	if createClient {
-		require.NoError(t, x.cl.CreateClient(context.Background(), cl))
+		require.NoError(t, x.Cl.CreateClient(context.Background(), cl))
 	}
 
-	require.NoError(t, x.co.CreateAuthenticationRequest(context.Background(), &consent.AuthenticationRequest{Client: cl, OpenIDConnectContext: new(consent.OpenIDConnectContext), Challenge: id, Verifier: id, AuthenticatedAt: time.Now(), RequestedAt: time.Now()}))
-	require.NoError(t, x.co.CreateConsentRequest(context.Background(), cr))
-	_, err := x.co.HandleConsentRequest(context.Background(), id, &consent.HandledConsentRequest{
+	require.NoError(t, x.Co.CreateAuthenticationRequest(context.Background(), &consent.AuthenticationRequest{Client: cl, OpenIDConnectContext: new(consent.OpenIDConnectContext), Challenge: id, Verifier: id, AuthenticatedAt: time.Now(), RequestedAt: time.Now()}))
+	require.NoError(t, x.Co.CreateConsentRequest(context.Background(), cr))
+	_, err := x.Co.HandleConsentRequest(context.Background(), id, &consent.HandledConsentRequest{
 		ConsentRequest: cr, Session: new(consent.ConsentRequestSessionData), AuthenticatedAt: time.Now(),
 		Challenge:   id,
 		RequestedAt: time.Now(),
 	})
 	require.NoError(t, err)
 }
 
-func testHelperUniqueConstraints(m managerTestSetup, storageType string) func(t *testing.T) {
+type ManagerTestSetup struct {
+	F  pkg.FositeStorer
+	Cl client.Manager
+	Co consent.Manager
+}
+
+func TestHelperRunner(t *testing.T, store ManagerTestSetup, k string) {
+	t.Helper()
```

Comment:
Marked this function as a `Helper` since its function name would be useless in any stack output when running tests.